queries/snakemake/highlights.scm

```diff
@@ -0,0 +1,61 @@
+; inherits: python
+
+;; Compound directives
+["rule" "checkpoint" "module"] @keyword
+
+;; Simple directives
+(directive name: _ @keyword)
```

Comment:
I don't think directives should be `@field`s because not all of them can be referenced during interpolation. For example, you can define a `message:` directive in a rule, but you cannot refer to this with `{message}` in the shell block. 
Also, directives themselves can have fields that can be referred to individually (`input.a`, `input.b`). You could even assign a dictionary to `input.a`, and you would later refer to its keys as `input.a.key1`, `input.a.key2`, ...
Any wildcard names are also available as fields (`wildcards.f`), which is why I chose `@field` for them.

In my opinion, it is important to highlight the elements in a wildcard interpolation (`input.a`) the same way as their definition `input: a = "..."`

Concerning changing directive names from `@keyword` to `@variable.builtin` -- there is also the case of the `run` directive, which is an alternative to `shell` that runs python code instead of a shell command. This is syntactically close to `if x: do()`, in which `if` is a keyword. There are also free-standing directives which can appear outside of a block, for example `configfile: "config.yaml"`, which I think is captured better by `@keyword`. More generally, the word-colon sequence is reserved for keywords in base Python. I think the breadth of functions fulfilled by these directives warrants their `@keyword` status.

![Screenshot_20230212_192628](https://user-images.githubusercontent.com/8322329/218329766-1c1dbc84-79e8-4c61-ab9a-14f4216b35a9.jpg)

Reply:
OK